octavia-cli/octavia_cli/headers.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+from dataclasses import dataclass
+from itertools import groupby
+from typing import Optional, List, Tuple
+
+import click
+import yaml
+
+from .apply.yaml_loaders import EnvVarLoader
+
+API_HEADERS_YAML_FILE_INSTRUCTION = """
+headers:
+  - name: Authorization
+    value: Basic dXNlcjpwYXNzd29yZA==
+  - name: Content-Type
+    value: application/json
+""".strip()
+
+
+@dataclass
+class ApplicationHeader:
+    name: Optional[str]
+    value: Optional[str]
+
+    def __post_init__(self):
+        self.is_valid = all([self.name, self.value])
+
+    def clean(self):
+        return ApplicationHeader(
+            name=self.name.strip(),
+            value=self.value.strip()
+        )
+
+
+def deduplicate_api_headers(application_headers: List[ApplicationHeader]) -> List[ApplicationHeader]:
+    sorted_headers = sorted(application_headers, key=lambda header: header.name)
+    groups = groupby(sorted_headers, key=lambda header: header.name)
+    unique_headers = [
+        ApplicationHeader(
+            name=header_name,
+            value=sorted([header.value for header in group_headers])[0]
+        ) for header_name, group_headers in groups]
+    return unique_headers
+
+
+def read_yaml_header_configuration(header_configuration_path: str) -> List[ApplicationHeader]:
+    with open(header_configuration_path) as file:
+        content = yaml.load(file, EnvVarLoader)
+
+    headers = content.get("headers", None)
+
+    if not headers:
+        raise AttributeError("""
+        Please provide valid configuration file, like below
+        headers:
+          - name: Authorization
+            value: Basic dXNlcjpwYXNzd29yZA==
+          - name: Content-Type
+            value: application/json
+        """)
+
+    application_headers = [
+        ApplicationHeader(header_pair.get("name", None), header_pair.get("value")) for header_pair in headers
+    ]
+
+    valid_headers = [
+        app_header.clean() for app_header in application_headers if app_header.is_valid
+    ]
+
+    return deduplicate_api_headers(valid_headers)
+
+
+def get_argument_based_headers(api_headers: List[Tuple[str, str]]) -> List[ApplicationHeader]:
+    valid_api_headers = [
+        ApplicationHeader(
+            name=api_header_name,
+            value=api_header_value
+        ).clean() for api_header_name, api_header_value in api_headers
+    ]
+
+    return deduplicate_api_headers(valid_api_headers)
```

Comment:
```suggestion
    return [
        ApplicationHeader(header_name, header_value) for header_name, header_value in api_headers
    ]
```
If you follow my `__post_init__` suggestion and `merge_api_headers` this function can be as simple as what I suggest.

Reply:
@Imbruced let me know what you think about this suggestion.